src/gui/DatabaseTabWidget.cpp

```diff
@@ -387,76 +349,31 @@ bool DatabaseTabWidget::saveDatabaseAs(Database* db)
-        if (!fileName.isEmpty()) {
-            QFileInfo fileInfo(fileName);
-            QString lockFilePath;
-            if (fileInfo.exists()) {
-                // returns empty string when file doesn't exist
-                lockFilePath = fileInfo.canonicalPath();
-            } else {
-                lockFilePath = fileInfo.absolutePath();
-            }
-            QString lockFileName = QString("%1/.%2.lock").arg(lockFilePath, fileInfo.fileName());
-            QScopedPointer<QLockFile> lockFile(new QLockFile(lockFileName));
-            lockFile->setStaleLockTime(0);
-            if (!lockFile->tryLock()) {
-                // for now silently ignore if we can't create a lock file
-                // due to lack of permissions
-                if (lockFile->error() != QLockFile::PermissionError) {
-                    QMessageBox::StandardButton result = MessageBox::question(this, tr("Save database as"),
-                        tr("The database you are trying to save as is locked by another instance of KeePassXC.\n"
-                        "Do you want to save it anyway?"),
-                        QMessageBox::Yes | QMessageBox::No);
-
-                    if (result == QMessageBox::No) {
-                        return false;
-                    } else {
-                        // take over the lock file if possible
-                        if (lockFile->removeStaleLockFile()) {
-                            lockFile->tryLock();
-                        }
-                    }
-                }
-            }
-
-            // setup variables so saveDatabase succeeds
-            dbStruct.saveToFilename = true;
-            dbStruct.canonicalFilePath = fileName;
+        if (!newFilePath.isEmpty()) {
+            // Ensure we don't recurse back into this function
+            dbStruct.readOnly = false;
 
-            if (!saveDatabase(db)) {
-                // failed to save, revert back
-                dbStruct.saveToFilename = false;
-                dbStruct.canonicalFilePath = oldFileName;
+            if (!saveDatabase(db, newFilePath)) {
+                // Failed to save, try again
                 continue;
             }
 
-            // refresh fileinfo since the file didn't exist before
-            fileInfo.refresh();
-
-            dbStruct.modified = false;
-            dbStruct.saveToFilename = true;
-            dbStruct.readOnly = false;
-            dbStruct.filePath = fileInfo.absoluteFilePath();
-            dbStruct.canonicalFilePath = fileInfo.canonicalFilePath();
-            dbStruct.fileName = fileInfo.fileName();
-            dbStruct.dbWidget->updateFilename(dbStruct.filePath);
-            delete dbStruct.lockFile;
-            dbStruct.lockFile = lockFile.take();
-            updateTabName(db);
-            updateLastDatabases(dbStruct.filePath);
+            dbStruct.dbWidget->updateFilename(dbStruct.fileInfo.absoluteFilePath());
```

Comment:
@droidmonkey maybe we can rename this `updateFilePath`?

Reply:
Absolutely, fixed in other places as well.